transcom/mymove
# B-22137 & B-22170 update pdfcpu 0.6.0 to latest 0.9.1

Current ticket [B-22170](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1050367)
[Originated from B-22137](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-22137)

## Summary
Production containers encountering stack overflow on pdfcpu call. 

Pdfcpu issue 867 was resolved in May of this year. Our current pdfcpu package `0.6.0` dates back to December 2023. Updating to latest `0.9.1` to attempt fixing issue.

### Docker changes
We now mount a separate volume in docker to not abide by the readonly aws ecs root marking a dedicated mutable directory. Pdfcpu uses this to install and write fonts because it doesn't do it on compile, it installs fonts and such after compile.

### Misc changes
Disabled watermarks in paperwork package generator. Watermarks never worked, they are using bookmark specific code. Enhanced by adding a pre-emptive nil check and then disabled it. The pre-emptive nil check will come in handy should a feature come through requesting watermarking.

Additionally, fixed some existing err return issues

## How to test
Anything that's got to do with milmove PDF generation, make sure it works! AOA/SSW/PPM you name it if it's PDF generated we want to make sure it works

Related
https://github.com/pdfcpu/pdfcpu/issues/867 <- May 2024
https://github.com/pdfcpu/pdfcpu/issues/268 <- December 2020
